content/en/blog/_posts/2021-11-02-csi-migration-status.md

```diff
@@ -0,0 +1,122 @@
+
+## What is the timeline/status?
+
+The current and targeted releases for each individual cloud driver is shown in the table below:
+
+| Driver            | Alpha         | Beta (in-tree deprecated) | Beta (on-by-default) | GA            | Target "in-tree plugin" removal |
+| ----------------- | ------------- | ------------------------- | -------------------- | ------------- | ------------------------------- |
+| AWS EBS           | 1.14          | 1.17                      | 1.23                 | 1.24 (Target) | 1.26 (Target)                   |
+| GCE PD            | 1.14          | 1.17                      | 1.23                 | 1.24 (Target) | 1.26 (Target)                   |
+| OpenStack Cinder  | 1.14          | 1.18                      | 1.21                 | 1.24 (Target) | 1.26 (Target)                   |
+| Azure Disk        | 1.15          | 1.19                      | 1.23                 | 1.24 (Target) | 1.26 (Target)                   |
+| Azure File        | 1.15          | 1.21                      | 1.23                 | 1.24 (Target) | 1.26 (Target)                   |
+| VSphere           | 1.18          | 1.19                      | 1.23 (Potentially miss)| 1.24 (Target) | 1.26 (Target)                   |
+| Ceph RBD          | 1.23          |
+| Portworx          | 1.23          |
+
+The following drivers will not have CSI migration support and will be deprecated and removed from the core Kubernetes.
+
+| Driver    | Deprecated | Code Removal  |
+| ----------| -----------| ------------- |
+| StorageOS | 1.22       | 1.25 (Target) |
+| ScaleIO   | 1.16       | 1.22          |
+| Flocker   | 1.22       | 1.25 (Target) |
+| Quobyte   | 1.22       | 1.25 (Target) |
+
+## What's next?
+
+With more CSI drivers come to Migration GA, we will soon mark the overall CSI Migration feature as GA. We are expecting cloud provider in-tree storage plugins code removal to happen by Kubernetes v1.26.
+
+## What should I do as a user?
+
+Note that all new features for the Kubernetes storage system (like volume snapshotting) will only be added to the CSI interface. Therefore, if you are starting up a new cluster, creating stateful applications for the first time, or require these new features we recommend using CSI drivers natively (instead of the in-tree volume plugin API). Follow the [updated user guides for CSI drivers](https://kubernetes-csi.github.io/docs/drivers.html) and use the new CSI APIs.
+
+However, if you choose to roll a cluster forward or continue using specifications with the legacy volume APIs, CSI Migration will ensure we continue to support those deployments with the new CSI drivers. However, if you want to leverage new features like snapshot, it will require a manual migration to translate the in-tree PV spec to CSI.
```

Comment:
What do you mean by "manual migration"?  Does that involve ensuring the reclaim policy is retain, deleting the PVC/PV objects, and re-importing volumes by static provisioning?

Reply:
Yes. I added another sentence to explain this.